src/compressions/Gzip.ts

```diff
@@ -41,7 +41,12 @@ export class GzipCompression extends Compression {
    * Returns gzip compression instance in closure.
    */
   public getCompression(): () => zlib.Gzip {
-    return (): zlib.Gzip => zlib.createGzip(this.compressionOptions);
+    return (): zlib.Gzip =>
```

Comment:
@strmer15 You don't need to set options directly to `zlib.createGzip` method. Make changes inside `selectCompression` method like:

```javascript
  private selectCompression(): void {
    const options: GzipOptions = {};

    if (this.options.gzipLevel !== undefined) {
      options.level = this.options.gzipLevel;
    }

    if (this.options.gzipMemoryLevel !== undefined) {
      options.memLevel = this.options.gzipMemoryLevel;
    }

    if (this.options.gzipStrategy !== undefined) {
      options.strategy = this.options.gzipStrategy;
    }
    this.compressionOptions = options;
  }
```

Reply:
@gios updated to use the `selectCompression` function instead, note that the CLI output does change to say 'level', etc. instead of 'gzipLevel' when printing the config.